File: back_end/src/main/java/com/wireless/model/vo/DashboardVO.java
package com.wireless.model.vo;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.math.BigDecimal;
import java.util.List;
import java.util.Map;

/**
 * 控制面板仪表盘 VO
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class DashboardVO {
    /** 当前温度 */
    private BigDecimal currentTemp;
    /** 当前湿度 */
    private BigDecimal currentHumidity;
    /** 当前烟雾浓度 */
    private BigDecimal currentSmoke;
    /** 设备状态 Map<deviceCode, status> */
    private Map<String, Integer> deviceStatus;
    /** 今日告警次数 */
    private Integer todayAlertCount;
    /** 今日语音指令次数 */
    private Integer todayVoiceCount;
}

File: Android软件/app/src/main/java/com/smarthome/app/core/model/DashboardVO.kt
package com.smarthome.app.core.model

import com.squareup.moshi.Json

data class DashboardVO(
    @Json(name = "currentTemp") val currentTemp: Double? = null,
    @Json(name = "currentHumidity") val currentHumidity: Double? = null,
    @Json(name = "currentSmoke") val currentSmoke: Double? = null,
    @Json(name = "deviceStatus") val deviceStatus: Map<String, Int>? = null,
    @Json(name = "todayAlertCount") val todayAlertCount: Int = 0,
    @Json(name = "todayVoiceCount") val todayVoiceCount: Int = 0
)
